packages/invoices/src/OutboundShipment/DetailView/modals/BatchesTable.tsx

```diff
@@ -155,37 +175,39 @@ export const BatchesTable: React.FC<BatchesTableProps> = ({
 
   const rowsWithoutPlaceholder = rows.filter(({ id }) => id !== 'placeholder');
 
+  const isRequestedPackSize = (packSize: number) =>
+    packSizeController.selected.value === -1 ||
+    packSize === packSizeController.selected.value;
+
   const allocatableRows = rowsWithoutPlaceholder
     .filter(
-      ({ onHold, availableNumberOfPacks }) =>
-        !onHold && availableNumberOfPacks > 0
+      ({ onHold, availableNumberOfPacks, packSize }) =>
+        !onHold && availableNumberOfPacks > 0 && isRequestedPackSize(packSize)
     )
     .sort(sortByExpiry);
 
-  const nonAllocatableRows = rowsWithoutPlaceholder.filter(
-    ({ onHold, availableNumberOfPacks }) =>
-      onHold || availableNumberOfPacks === 0
-  );
-
-  const onHoldRows = nonAllocatableRows
+  const onHoldRows = rowsWithoutPlaceholder
     .filter(({ onHold }) => onHold)
     .sort(sortByExpiry);
 
-  const noStockRows = nonAllocatableRows
+  const noStockRows = rowsWithoutPlaceholder
     .filter(
       ({ availableNumberOfPacks, onHold }) =>
         availableNumberOfPacks === 0 && !onHold
     )
     .sort(sortByExpiry);
 
+  const wrongPackSizeRows = rowsWithoutPlaceholder
+    .filter(({ packSize }) => !isRequestedPackSize(packSize))
+    .sort(sortByExpiry);
+
   return (
     <>
       <Divider margin={10} />
-      <TableContainer sx={{ height: 400, overflowX: 'hidden' }}>
+      <TableContainer sx={{ height: 375, overflowX: 'hidden' }}>
```

Comment:
was getting two vertical scrollbars, so have reduced the table container height a little